test/entity.ts

```diff
@@ -744,32 +744,77 @@ describe('entity', () => {
       assert.deepStrictEqual(entity.encodeValue(value), expectedValueProto);
     });
 
-    it('should encode an int', () => {
+    it('should encode an int', done => {
       const value = 8;
 
       const expectedValueProto = {
         integerValue: value,
       };
 
+      const property = 'undefined';
+      const expectedWarning =
+        "TypeCastWarning: the value for '" +
+        property +
+        "' property is a JavaScript Number.\n" +
+        "Use 'Datastore.int(<integer_value_as_string>)' or " +
+        "'Datastore.double(<double_value_as_string>)' to preserve consistent " +
+        'Datastore types in your database.';
+      process.once('warning', warning => {
+        assert.strictEqual(warning.message, expectedWarning);
+        done();
+      });
+
+      entity.Int = function (value_: {}) {
```

Comment:
The monkey-patch here weirds me a bit. If Node is doing any kind of module caching, then the `require()` in `beforeEach()` might not give you a new copy each time. (Which might be the source of the test failures.) It might be safer to use Sinon stubs.

Reply:
I think this is why we delete it from the cache?

Reply:
Oh geez, I somehow missed that.